src/core/meshcop/tcat_agent.cpp

```diff
@@ -506,45 +514,106 @@ Error TcatAgent::HandleSingleTlv(const Message &aIncomingMessage, Message &aOutg
 
 Error TcatAgent::HandleSetActiveOperationalDataset(const Message &aIncomingMessage, uint16_t aOffset, uint16_t aLength)
 {
-    Dataset     dataset;
-    OffsetRange offsetRange;
-    Error       error;
+    Dataset       dataset;
+    OffsetRange   offsetRange;
+    Error         error;
+    unsigned char buf[kCommissionerCerMaxLength];
+    size_t        bufLen = sizeof(buf);
 
     offsetRange.Init(aOffset, aLength);
     SuccessOrExit(error = dataset.SetFrom(aIncomingMessage, offsetRange));
     SuccessOrExit(error = dataset.ValidateTlvs());
 
-    if (!CheckCommandClassAuthorizationFlags(mCommissionerAuthorizationField.mApplicationFlags,
-                                             mDeviceAuthorizationField.mApplicationFlags, &dataset))
+    if (!CheckCommandClassAuthorizationFlags(mCommissionerAuthorizationField.mCommissioningFlags,
+                                             mDeviceAuthorizationField.mCommissioningFlags, &dataset))
     {
         error = kErrorRejected;
         ExitNow();
     }
 
     Get<ActiveDatasetManager>().SaveLocal(dataset);
 
+    SuccessOrExit(error = otBleSecureGetPeerCertificateDer(&GetInstance(), buf, &bufLen));
```

Comment:
Usually, the API functions are not called from within the stack/core - only by the application (e.g. CLI) making use of the C API.  Internally, the corresponding C++ call is made directly.
(@abtink to confirm?)

Reply:
fixed.